packages/meteor/dynamics_nodejs.js

```diff
@@ -15,6 +14,7 @@ const SLOT_CALL_KEY = "slotCall";
 class EnvironmentVariableAsync {
   constructor() {
     this.slot = nextSlot++;
+    this.als = new AsyncLocalStorage()
```

Comment:
I am trying to understand if that would be preferrable compared to having complex logic in `bindEnvironment`... if CPU is a little worse and the event loop is fine then it would be great. Also, Node.js is going to bring optimizations for ALS in the next releases which might help in this scenario.

Reply:
> compared to having complex logic in bindEnvironment

The implementation is overly complex. I created #13425 to get it down to one piece of state stored in ALS.